Source/UINavigation/Private/UINavWidget.cpp

```diff
@@ -3148,11 +3148,12 @@ int UUINavWidget::GetGridStartingIndex(const int GridIndex)
 
 UUINavButton * UUINavWidget::GetButtonAtGridIndex(const int GridIndex, int IndexInGrid)
 {
-	if (!NavigationGrids.IsValidIndex(GridIndex) || IndexInGrid < 0) return nullptr;
+	if (!NavigationGrids.IsValidIndex(GridIndex)) return nullptr;
 
 	const FGrid& ButtonGrid = NavigationGrids[GridIndex];
 	if (ButtonGrid.FirstButton == nullptr) return nullptr;
 	if (IndexInGrid == -1) IndexInGrid = ButtonGrid.GetDimension() - 1;
+	if (IndexInGrid < 0) return nullptr;
```

Comment:
TL;DR: Remove this line.

This check makes sense, but this shouldn't happen and if it does I need the game to crash so people tell me something's wrong.
In case the grid has no buttons, the FirstButton will be a nullptr (this is checked 2 lines above) so this check is redundant (again, unless something else is wrong, in which case **that** should be fixed)

Reply:
The check for IndexInGrid < 0 was added in this recent commit:

https://github.com/goncasmage1/UINavigation/commit/c8888e2a1a1e7ca253e9c44413790a9498b9da1d

If you decide to remove it everything should be as before and work properly. Function is blueprint callable, from .h:

	/**
	*	Returns the button at the specified index of the given grid.
	*	Pass -1 for IndexInGrid for last button in grid
	*/
	UFUNCTION(BlueprintCallable, BlueprintPure, Category = UINavWidget)
	class UUINavButton* GetButtonAtGridIndex(const int GridIndex, int IndexInGrid);